docs/source/en/model_doc/audioflamingo3.md

```diff
@@ -0,0 +1,393 @@
+
+conversation = [
+    {
+        "role": "user",
+        "content": [
+            {"type": "text", "text": "What is the capital of France?"},
+        ],
+    }
+]
+
+batch = processor.apply_chat_template(
+    conversation,
+    tokenize=True,
+    add_generation_prompt=True,
+    return_dict=True,
+).to(model.device)
+
+gen_ids = model.generate(**batch)
+
+inp_len = batch["input_ids"].shape[1]
+new_tokens = gen_ids[:, inp_len:]
+texts = processor.batch_decode(new_tokens, skip_special_tokens=True)
+print(texts)
+```
+
+➡️ audio only:
+
+```python
+from transformers import AudioFlamingo3ForConditionalGeneration, AutoProcessor
+
+MODEL_ID = "nvidia/audio-flamingo-3-hf"
+processor = AutoProcessor.from_pretrained(MODEL_ID)
+model = AudioFlamingo3ForConditionalGeneration.from_pretrained(MODEL_ID, device_map="auto")
+
+conversation = [
+    {
+        "role": "user",
+        "content": [
+            {"type": "audio", "path": "https://audioflamingo3.github.io/static/chat/WhDJDIviAOg_120_10.mp3"},
+        ],
+    }
+]
+
+batch = processor.apply_chat_template(
+    conversation,
+    tokenize=True,
+    add_generation_prompt=True,
+    return_dict=True,
+).to(model.device)
+
+gen_ids = model.generate(**batch)
+
+inp_len = batch["input_ids"].shape[1]
+new_tokens = gen_ids[:, inp_len:]
+texts = processor.batch_decode(new_tokens, skip_special_tokens=True)
+print(texts)
+```
+
+➡️ batched inference!
+
+```python
+from transformers import AudioFlamingo3ForConditionalGeneration, AutoProcessor
+
+MODEL_ID = "nvidia/audio-flamingo-3-hf"
+processor = AutoProcessor.from_pretrained(MODEL_ID)
+model = AudioFlamingo3ForConditionalGeneration.from_pretrained(MODEL_ID, device_map="auto")
+
+conversations = [
+    [
+        {
+            "role": "user",
+            "content": [
+                {"type": "text", "text": "Transcribe the input speech."},
+                {
+                    "type": "audio",
+                    "path": "https://audioflamingo3.github.io/static/long_speech/t_837b89f2-26aa-4ee2-bdf6-f73f0dd59b26.wav",
+                },
+            ],
+        }
+    ],
+    [
+        {
+            "role": "user",
+            "content": [
+                {
+                    "type": "text",
+                    "text": "This track feels really peaceful and introspective. What elements make it feel so calming and meditative?",
+                },
+                {"type": "audio", "path": "https://audioflamingo3.github.io/static/chat/FPSbCAANfbJLVSwD.mp3"},
+            ],
+        }
+    ],
+]
+
+batch = processor.apply_chat_template(
+    conversations,
+    tokenize=True,
+    add_generation_prompt=True,
+    return_dict=True,
+).to(model.device)
+
+gen_ids = model.generate(**batch)
+
+inp_len = batch["input_ids"].shape[1]
+new_tokens = gen_ids[:, inp_len:]
+texts = processor.batch_decode(new_tokens, skip_special_tokens=True)
+print(texts)
+```
+
+➡️ transcription shortcut
```

Comment:
great, thanks for adding this example! 

@eustlb just want to double-check that this API looks good? (related to offline discussion we had with Lysandre) 

We went for something similar to Voxtral: https://github.com/huggingface/transformers/blob/b9f90dc388fd415a2ba2a6a31a372f451d4a4eed/src/transformers/models/voxtral/processing_voxtral.py#L278